File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        
        List<Integer> list = Arrays.stream(people)
                .boxed()
                .sorted(Comparator.reverseOrder())
                .collect(Collectors.toList());

        int first = 0;
        int last = list.size() - 1;

        while(first <= last) {
            if(list.get(first) + list.get(last) <= limit) {
                first++;
                last--;
            } else {
                first++;
            }
            answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.py
def solution(people, limit):
    answer = 1
    people.sort()
    left = 0
    right = len(people) - 1
    weight = 0
    count = 0
    while left <= right:
        if weight + people[right] <= limit and count < 2:
            weight += people[right]
            count += 1
            right -= 1
        elif weight + people[left] <= limit and count < 2:
            weight += people[left]
            count += 1
            left += 1
        else:
            weight = 0
            count = 0
            answer += 1

        
    return answer